app/views/RoomView/List/index.tsx

```diff
@@ -31,367 +17,64 @@ const styles = StyleSheet.create({
-	}
-
-	// clears previous highlighted message timeout, if exists
-	clearHighlightedMessageTimeout = () => {
-		if (this.highlightedMessageTimeout) {
-			clearTimeout(this.highlightedMessageTimeout);
-			this.highlightedMessageTimeout = false;
-		}
-	};
-
-	query = async () => {
-		this.count += QUERY_SIZE;
-		const { rid, tmid, showMessageInMainThread, serverVersion } = this.props;
-		const db = database.active;
-
-		// handle servers with version < 3.0.0
-		let { hideSystemMessages = [] } = this.props;
-		if (!Array.isArray(hideSystemMessages)) {
-			hideSystemMessages = [];
-		}
-
-		if (tmid) {
-			try {
-				this.thread = await db.get('threads').find(tmid);
-			} catch (e) {
-				console.log(e);
-			}
-			this.messagesObservable = db
-				.get('thread_messages')
-				.query(Q.where('rid', tmid), Q.experimentalSortBy('ts', Q.desc), Q.experimentalSkip(0), Q.experimentalTake(this.count))
-				.observe();
-		} else if (rid) {
-			const whereClause = [
-				Q.where('rid', rid),
-				Q.experimentalSortBy('ts', Q.desc),
-				Q.experimentalSkip(0),
-				Q.experimentalTake(this.count)
-			] as (Q.WhereDescription | Q.Or)[];
-			if (!showMessageInMainThread) {
-				whereClause.push(Q.or(Q.where('tmid', null), Q.where('tshow', Q.eq(true))));
-			}
-			this.messagesObservable = db
-				.get('messages')
-				.query(...whereClause)
-				.observe();
-		}
-
-		if (rid) {
-			this.unsubscribeMessages();
-			this.messagesSubscription = this.messagesObservable?.subscribe(messages => {
-				if (tmid && this.thread) {
-					messages = [...messages, this.thread];
-				}
-
-				/**
-				 * Since 3.16.0 server version, the backend don't response with messages if
-				 * hide system message is enabled
-				 */
-				if (compareServerVersion(serverVersion, 'lowerThan', '3.16.0') || hideSystemMessages.length) {
-					messages = messages.filter(m => !m.t || !hideSystemMessages?.includes(m.t));
-				}
-
-				if (this.mounted) {
-					this.setState({ messages }, () => this.update());
-				} else {
-					// @ts-ignore
-					this.state.messages = messages;
-				}
-				// TODO: move it away from here
-				this.readThreads();
-			});
-		}
-	};
-
-	reload = () => {
-		this.count = 0;
-		this.query();
-	};
-
-	readThreads = debounce(async () => {
-		const { tmid } = this.props;
-
-		if (tmid) {
-			try {
-				await Services.readThreads(tmid);
-			} catch {
-				// Do nothing
-			}
-		}
-	}, 300);
-
-	onEndReached = () => this.query();
-
-	onRefresh = () =>
-		this.setState({ refreshing: true }, async () => {
-			const { messages } = this.state;
-			const { rid, tmid } = this.props;
-
-			if (messages.length) {
-				try {
-					if (tmid) {
-						await loadThreadMessages({ tmid, rid });
-					} else {
-						await loadMissedMessages({ rid, lastOpen: moment().subtract(7, 'days').toDate() });
-					}
-				} catch (e) {
-					log(e);
-				}
-			}
-
-			this.setState({ refreshing: false });
+const ListContainer = forwardRef<IListContainerRef, IListContainerProps>(
```

Comment:
I didn't want to do it, but following the index pattern, I should've created `ListContainer.tsx` and use `index.ts` only for export.
Let me know if I should do it.